### PR TITLE
feat: Allow Docker container to run as root with builtin pg as non-root

### LINKED ISSUE #5263: Quickstart Docker on macOS
Hi guys,

According to [quickstart/docker](https://coder.com/docs/coder-oss/latest/quickstart/docker) it should be possible to get started with macOS & docker. Which I would really appreciate to get a first look of what coder is and how I could use it.

Sadly the docu seems to be Linux specific - for example `export DOCKER_GROUP=$(getent group docker | cut -d: -f3)` will not help me under macOS.
And running the docker command without `--group-add $DOCKER_GROUP` will net me the error `Error pinging Docker server: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock:` in step `coder templates create`. Same result when trying the docker-compose approach.
Yes in the troubleshooting section is a link to [Manage Docker as a non-root user](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user) - which does not really help under macOS...

How would I get started with a coder docker deployment under macOS? 

Sorry for the noob questions,
Felix

### PR DESCRIPTION
**Pull request by mafredri**
This is a prototype implementation for allowing to run embedded postgres with lower credentials inside Docker if started as root.

If we go forward, this needs to be cleaned up.

**Purpose:** Allow Docker Desktop users on macOS to quick-start via:

```
docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock --name coder --user root ghcr.io/coder/coder:latest
```

This would allow us to improve documentation and fix https://github.com/coder/coder/issues/5263.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,6 +22,10 @@ COPY --chown=1000:1000 --chmod=700 empty-dir /home/coder
 # The coder binary is injected by scripts/build_docker.sh.
 COPY --chown=1000:1000 --chmod=755 coder /opt/coder
 
+# For cases where the container is run as root, allow embedded postgres,
+# if enabled, to be de-elevated.
+ENV CODER_CONFIG_DIR=/home/coder/.config/coderv2 CODER_BUILTIN_POSTGRES_UID=1000 CODER_BUILTIN_POSTGRES_GID=1000
```
Comment (deansheather) [inline Dockerfile:27]: For clarity I think these should be 3 separate ENV lines. ENV statements don't create layers in the resulting image AFAIK.

File: cli/server.go [hunk 2]
```diff
@@ -187,17 +189,106 @@ func Server(vip *viper.Viper, newAPI func(context.Context, *coderd.Options) (*co
...
+				// e.g. the Docker socket.
+				case u.Uid == "0":
+					parseUint32 := func(s string) uint32 {
+						v, err := strconv.ParseUint(s, 10, 32)
+						if err != nil {
+							return 0
+						}
+						return uint32(v)
+					}
+					uid := parseUint32(os.Getenv("CODER_BUILTIN_POSTGRES_UID"))
```
Comment (deansheather) [inline cli/server.go:209]: should be done through deploymentconfig

**Review by deansheather — COMMENTED** [review]
Looks good! I don't think we should bother supporting windows for this though as per our discussion on slack

**Review by jsjoeio — COMMENTED** [review]
This sounds like a great UX improvement! 👏🏼

**Pull request closed without merging**